documents/README.md

```diff
@@ -0,0 +1,176 @@
+
+```ts
+// src/factory.ts#L32-L40
+
+  protected createNewWidget(
+    context: DocumentRegistry.IContext<ExampleDocModel>
+  ): ExampleDocWidget {
+    return new ExampleDocWidget({
+      context,
+      content: new ExamplePanel(context)
+    });
+  }
+}
+```
+
+On the other hand, to create a `ModelFactory`, you need to implement the interface `IModelFactory<T>` specifying the name of your model, which type of files represents and its format.
+
+```ts
+// src/factory.ts#L48-L67
+```
+
+At the same time, you need to implement the method `createNew`. The `DocumentManager` will call this method when opening a file that uses your custom `DocumentModel`.
+
+```ts
+// src/factory.ts#L92-L101
+```
+
+## Registering new Documents
+
+When registering a new document, the first step is to know which file type you want to add a new `DocumentModel`. If the file type is already registered, you won't need to register it again. You could register a new `DocumentModel` for an existing file type. If the file type you want to support is not registered, you will need to register it. To do that, you can use the API `addFileType` from the `DocumentRegistry`. The essential arguments are `extensions` to indicate the extension of the file, `fileFormat` that specifies the data format, and `contentType` to determine if it is a notebook, file or directory.
+
+```ts
+// src/index.ts#L73-L81
+
+// register the filetype
+app.docRegistry.addFileType({
+  name: 'example',
+  displayName: 'Example',
+  mimeTypes: ['text/json', 'application/json'],
+  extensions: ['.example'],
+  fileFormat: 'text',
+  contentType: 'file'
+});
+```
+
+Once the file type is registered, you can register a `DocumentModel` for a specific file type. The `DocumentModel` represent the content of the file. For example, JupyterLab has two models registered for the notebook. When you open a notebook with the Notebook editor, the `DocumentManager` creates an instance of the `NotebookModel` that loads the notebook as a JSON object and offers a complex API to manage cells and metadata independently (treats the content of the notebook as a structured data), opposite when opening a notebook with the plain text editor the `DocumentManager` creates an instance of the base `DocumentModel` class which treats the content of the notebook as a string. Note that you can register multiple models for the same file type. Still, these models are not in sync when the user opens two editors for the same file that use different models (like opening a notebook with the notebook editor and the plain text editor). These editors are not in sync because they use other models. At some point, they will show different content.
```

Comment:
```suggestion
Once the file type is registered, you can register a `DocumentModel` for a specific file type. The `DocumentModel` represents the content of the file. For example, JupyterLab has two models registered for the notebook. When you open a notebook with the Notebook editor, the `DocumentManager` creates an instance of the `NotebookModel` that loads the notebook as a JSON object and offers a complex API to manage cells and metadata independently (treats the content of the notebook as a structured data). When opening a notebook with the plain text editor the `DocumentManager` creates an instance of the base `DocumentModel` class which treats the content of the notebook as a string. Note that you can register multiple models for the same file type. Still, these models are not in sync when the user opens two editors for the same file that use different models (like opening a notebook with the notebook editor and the plain text editor). These editors are not in sync because they use different models. At some point, they will show different content.
```